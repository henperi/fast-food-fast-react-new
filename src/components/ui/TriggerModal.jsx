import React, { Component } from 'react';
import Modal from './Modal/Modal';

export class TriggerModal extends Component {
  state = {
    modal: {
      id: undefined,
      isOpen: false,
    },
  };

  handleModal = (
    event,
    {
      id = '', type = '', title = '', price = '', description = '',
    } = this.props,
  ) => {
    if (event && event.target.classList.contains('triggerModal')) {
      return this.setState(() => ({
        modal: {
          isOpen: true,
          id,
          title,
          type,
          price,
          description,
        },
      }));
    }
    if (event && event.target.classList.contains('close-button')) {
      return this.setState(() => ({
        modal: {
          isOpen: false,
        },
      }));
    }
    return this.setState(() => ({
      modal: {
        isOpen: false,
      },
    }));
  };

  render() {
    const { triggerClass, triggerName } = this.props;
    const { modal } = this.state;

    return (
      <span>
        <button onClick={this.handleModal} type="button" className={`triggerModal ${triggerClass}`}>
          {triggerName}
        </button>
        <Modal {...modal} handleModal={this.handleModal} {...this.props} />
      </span>
    );
  }
}

export default TriggerModal;
