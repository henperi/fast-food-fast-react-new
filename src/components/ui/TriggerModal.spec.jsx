import React from 'react';
import { shallow } from 'enzyme';
import TriggerModal from './TriggerModal';

test('should render TriggerModal correctly', () => {
  const wrapper = shallow(<TriggerModal />);
  expect(wrapper).toMatchSnapshot();
});
