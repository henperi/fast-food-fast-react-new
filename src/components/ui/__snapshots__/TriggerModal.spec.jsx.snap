// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TriggerModal correctly 1`] = `
<span>
  <button
    className="triggerModal undefined"
    onClick={[Function]}
    type="button"
  />
  <modal
    handleModal={[Function]}
    isOpen={false}
  />
</span>
`;
